src/main/java/com/salesforce/cdp/queryservice/util/Utils.java

```diff
@@ -25,10 +25,15 @@
 
 public class Utils {
 
+    // NOTE: SC_UNPROCESSABLE_ENTITY added for retry due to bug W-9768558. Would be removed once this bug is fixed.
     private static Set<Integer> retryStatusCodes = new HashSet<>(Arrays.asList(ExtendedHttpStatusCode.SC_TOO_MANY_REQUESTS,
             ExtendedHttpStatusCode.SC_MOVED_TEMPORARILY,
             ExtendedHttpStatusCode.SC_GATEWAY_TIMEOUT,
-            ExtendedHttpStatusCode.SC_SERVICE_UNAVAILABLE));
+            ExtendedHttpStatusCode.SC_SERVICE_UNAVAILABLE,
+            ExtendedHttpStatusCode.SC_UNPROCESSABLE_ENTITY,
+            ExtendedHttpStatusCode.SC_UNAUTHORIZED,
+            ExtendedHttpStatusCode.SC_INTERNAL_SERVER_ERROR,
+            ExtendedHttpStatusCode.SC_BAD_REQUEST));
```

Comment:
I am not sure if this treats getting core token and offcore token as a single unit, meaning if the cdp token exchange fails it will only retry that, but what we want is to get a new core token and then retry the cdp token exchange